crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -122,6 +124,22 @@ impl SuiEvent {
     }
 }
 
+impl Display for SuiEvent {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let output = format!(
+            " ┌──\n │ EventID: {}\n │ PackageID: {}\n │ Transaction Module: {}\n │ Sender: {} \n │ EventType: {}:{}\n │ ParsedJSON: {}\n │ BCS: {:?}\n",
+            self.id.tx_digest, self.id.event_seq, self.package_id, self.transaction_module, self.sender, self.type_, self.parsed_json, self.bcs);
```

Comment:
Also, we shouldn't create a `String` using `format!` only to put it into another format string, you could be writing this directly into `f`:
```suggestion
        write!(
            f,
            " ┌──\n │ EventID: {}\n │ PackageID: {}\n │ Transaction Module: {}\n │ Sender: {} \n │ EventType: {}:{}\n │ ParsedJSON: {}\n │ BCS: {:?}\n",
            self.id.tx_digest, self.id.event_seq, self.package_id, self.transaction_module, self.sender, self.type_, self.parsed_json, self.bcs)?;
```